server/app/services/geo/AddressSuggestion.java

```diff
@@ -43,5 +44,33 @@ public abstract static class Builder {
     public abstract Builder setAddress(Address address);
 
     public abstract AddressSuggestion build();
+
+    public AddressSuggestion fromJson(JsonNode node) {
```

Comment:
Suggestion (blocking): This could be removed if the address correction and location fields were added to the form itself.

Reply:
Keeping the use of session. Resolving.